### DISCUSSION TITLE
Wild bootstrap for inference on clustered data

### DISCUSSION THREAD

**Discussion by Sanford-Lab** [DoubleML/doubleml-for-r#190] [Q&A]
Hi all,
Looking to conduct inference where we have a small number of large clusters and want to get valid confidence intervals. I know of implementations of (multiplier) bootstrap methods for simultaneous inference and the cluster-robust inference that has been implemented, wondering if those two pieces play well together to get bootstrap CIs for clustered data.
Thanks!

**Comment by PhilippBach** [comment]
Hi @Sanford-Lab , 

thanks for your excellent question! I am not 100% sure about the interaction of the multiplier bootstrap with clustered data. As of now, we haven't implemented a bootstrap method that takes the clustering of the data into account... see https://github.com/DoubleML/doubleml-for-r/blob/ba452aba8f6c71df2da560e6d009fcb565079dd6/R/double_ml.R#L449 ; but I see that this is a relevant point, so I think we can add it as a new feature request. Before we'd implement it, we'd also have to check the literature on any papers covering this topic. 
In case you make some progress in that regard, please post any resource below here. And of course, you are cordially invited to add your changes to the DoubleML repo, please have a look at [our contributing guide](https://github.com/DoubleML/doubleml-for-r/blob/main/CONTRIBUTING.md)

Thanks again and all the best!

**Comment by inkrement** [comment]
@PhilippBach Hi! Is there any plan to support clustered bootstrapping in near future?

**Comment by PhilippBach** [comment]
Hi @inkrement , 

we'll work on this soon. Probably it will first be integrated in the python version https://github.com/DoubleML/doubleml-for-py
